### ISSUE TITLE
Error when quickly navigating between pages where the gridster is used.

### ISSUE THREAD

**Issue by dbas-dn** [tiberiuzuld/angular-gridster2#802]
Error description:
```
zone.js:218 Uncaught TypeError: Cannot read properties of undefined (reading 'unsubscribe')
    at GridsterPreviewComponent.ngOnDestroy (angular-gridster2.js:822:18)
    at executeOnDestroys (core.js:7418:32)
    at cleanUpView (core.js:7321:9)
    at destroyViewTree (core.js:7154:17)
    at destroyLView (core.js:7299:9)
    at ViewContainerRef.remove (core.js:23257:13)
    at ViewContainerRef.clear (core.js:23165:18)
    at NgIf._updateView (common.js:3459:37)
    at NgIf.set (common.js:3436:14)
    at setInputsForProperty (core.js:10979:13)
    at elementPropertyInternal (core.js:10023:9)
    at ɵɵproperty (core.js:14806:9)
    at PagesComponent_Template (template.html:17:32)
    at executeTemplate (core.js:9618:9)
    at refreshView (core.js:9484:13)
    at refreshComponent (core.js:10655:13)
    at refreshChildComponents (core.js:9280:9)
    at refreshView (core.js:9534:13)
    at refreshEmbeddedViews (core.js:10609:17)
    at refreshView (core.js:9508:9)
    at detectChangesInternal (core.js:10837:9)
    at ViewRef.detectChanges (core.js:22839:9)
```

https://github.com/tiberiuzuld/angular-gridster2/blob/fea8bfe3eaea459a979df62ec887f1e7a2e727f8/projects/angular-gridster2/src/lib/gridsterPreview.component.ts#L27

very strange situation. Please add a question mark to avoid exceptions.
```
ngOnDestroy(): void {
    this.sub?.unsubscribe();
    // @ts-ignore
    delete this.el;
  }
```

Dependencies:
```
├── @angular-devkit/build-angular@12.2.17
├── @angular/animations@12.2.16
├── @angular/cdk-experimental@12.2.13
├── @angular/cdk@12.2.13
├── @angular/cli@12.2.17
├── @angular/common@12.2.16
├── @angular/compiler-cli@12.2.16
├── @angular/compiler@12.2.16
├── @angular/core@12.2.16
├── @angular/flex-layout@12.0.0-beta.35
├── @angular/forms@12.2.16
├── @angular/material-moment-adapter@12.2.13
├── @angular/material@12.2.13
├── @angular/platform-browser-dynamic@12.2.16
├── @angular/platform-browser@12.2.16
├── @angular/router@12.2.16
├── @angularclass/hmr@3.0.0
├── @danielmoncada/angular-datetime-picker@12.3.0
├── @elfalem/leaflet-curve@0.8.2
├── @fortawesome/fontawesome-free@5.15.4
├── @ng-select/ng-select@7.4.0
├── @ngneat/until-destroy@8.1.4
├── @ngx-translate/core@13.0.0
├── @ngx-translate/http-loader@6.0.0
├── @pebula/ngrid-material@4.0.0
├── @pebula/ngrid@4.0.0
├── @swimlane/ngx-charts@14.0.0
├── @swimlane/ngx-graph@7.2.0
├── @types/chart.js@2.9.36
├── @types/d3@5.16.4
├── @types/file-saver@2.0.5
├── @types/jasmine@2.8.19
├── @types/leaflet.markercluster@1.4.6
├── @types/leaflet@1.7.9
├── @types/lodash@4.14.181
├── @types/resumablejs@1.1.0
├── @types/sprintf-js@1.1.2
├── @types/ssh2@0.5.52
├── angular-calendar@0.28.28
├── angular-gridster2@12.1.1
├── angular-resize-event@2.1.0
├── angular-split@4.0.0
├── apexcharts@3.35.0
├── async@3.2.3
├── bluebird@3.7.2
├── chart.js@2.9.4
├── chroma-js@1.4.1
├── codelyzer@5.2.2
├── core-js@3.21.1
├── d3-array@2.12.1
├── d3-brush@1.1.6
├── d3-color@1.4.1
├── d3-format@1.4.5
├── d3-hierarchy@1.1.9
├── d3-interpolate@1.4.0
├── d3-scale@3.3.0
├── d3-selection@1.4.2
├── d3-shape@1.3.7
├── d3-time-format@2.3.0
├── d3-transition@1.3.2
├── d3@5.16.0
├── date-fns@2.28.0
├── file-saver@2.0.5
├── http-status-codes@1.4.0
├── jasmine-core@2.99.1
├── jasmine-spec-reporter@4.2.1
├── leader-line@1.0.7
├── leaflet-map@0.2.1
├── leaflet.featuregroup.subgroup@1.0.2
├── leaflet.markercluster.layersupport@2.0.1
├── leaflet.markercluster@1.5.3
├── leaflet.tilelayer.colorfilter@1.2.5
├── leaflet@1.7.1
├── libphonenumber-js@1.9.50
├── lodash@4.17.21
├── long@4.0.0
├── moment-range@4.0.2
├── moment@2.29.2
├── ng-apexcharts@1.5.1
├── ng-inline-svg@11.0.2
├── ng-pick-datetime@7.0.0
├── ng-terminal@2.4.0
├── ng2-charts@2.3.3
├── ng2-file-upload@1.4.0
├── ngx-color-picker@11.0.0
├── ngx-ip@1.1.4
├── ngx-json-viewer@2.4.0
├── ngx-mask@11.0.1
├── ngx-perfect-scrollbar@10.1.1
├── ngx-pipes@2.7.8
├── ngx-scrollbar@8.0.0
├── normalize.css@8.0.1
├── object-hash@2.2.0
├── protobufjs@6.11.2
├── protractor@5.1.0
├── resize-observer-polyfill@1.5.1
├── resumablejs@1.1.0
├── rimraf@3.0.0
├── roboto-fontface@0.10.0
├── rxjs@6.5.4
├── socket.io-client@4.4.1
├── sprintf-js@1.1.2
├── standard-changelog@2.0.27
├── string-hash@1.1.3
├── stylelint@13.13.1
├── testcafe@1.18.5
├── three-orbitcontrols-ts@0.1.2
├── three@0.130.1
├── ts-morph@4.3.3
├── tslib@2.3.1
├── tslint-eslint-rules@5.4.0
├── tslint@5.20.1
├── typescript-tslint-plugin@0.5.5
├── typescript@4.2.4
├── url-parse@1.5.10
├── vis@4.21.0-EOL
└── zone.js@0.11.5
```

**Comment by Vasanthvivi** [comment]
Everything maintained by gridster itself,, you no need to care on resources

**Comment by tiberiuzuld** [comment]
Fix merged in v15.0.2
